nixos/modules/system/boot/systemd/sysusers.nix

```
@@ -89,7 +89,7 @@ in
         }
       ]
       ++ (lib.mapAttrsToList (username: opts: {
-        assertion = !opts.isNormalUser;
+        assertion = opts.enable && !opts.isNormalUser;
```

`&&` means that *any* user with `enable = false;` will trip this assertion. i.e. `{ enable = false; isSystemUser = true; }` trips it. It should be the implication operator instead.
```suggestion
        assertion = opts.enable -> !opts.isNormalUser;
```